File: app/src/main/java/com/osapps/capitalslearnproject/fragmentactivity/presentation/DetailPresenterImpl.java
package com.osapps.capitalslearnproject.fragmentactivity.presentation;

import com.osapps.capitalslearnproject.fragmentactivity.view.DetailView;
import com.osapps.capitalslearnproject.main.presentation.ItzikInstance;

import javax.inject.Inject;


/**
 * Created by mertsimsek on 30/05/2017.
 */

public class DetailPresenterImpl implements DetailPresenter{

    DetailView detailView;
    ItzikInstance instance;

    @Inject
    public DetailPresenterImpl(DetailView detailView, ItzikInstance instance) {
        this.detailView = detailView;
        this.instance = instance;
    }

    public void loadDetail(){
        instance.talkItzik();
        detailView.onDetailLoaded();
    }
}

File: app/src/main/java/com/osapps/capitalslearnproject/fragmentactivity/presentation/DetailPresenterImpl.kt
package com.osapps.capitalslearnproject.fragmentactivity.presentation

import com.osapps.capitalslearnproject.fragmentactivity.view.DetailView
import com.osapps.capitalslearnproject.main.presentation.ItzikInstance

import javax.inject.Inject


/**
 * Created by osApps on 30/05/2017.
 */

class DetailPresenterImpl @Inject
constructor(private var detailView: DetailView, internal var instance: ItzikInstance) : DetailPresenter {

    override fun loadDetail() {
        instance.talkItzik()
        detailView.onDetailLoaded()
    }
}
